### PR TITLE
surelog: init at 1.35

### PR DESCRIPTION
**Pull request by hzeller**
###### Description of changes

New package.

Surelog is a SystemVerilog 2017 Pre-processor, Parser,
Elaborator, and UHDM Compiler.

URL: https://github.com/chipsalliance/Surelog

It is used to parse the hardware description language
SystemVerilog and provide an intermediate format (UHDM) that
then can be used in other tools that need SystemVerilog
support for FPGAs and ASICs. There is a plug-in for Yosys
that can use it (will package separately)

Part of an effort to get more hardware development software readily
installable in nixos.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable
     - included `checkPhase`, that runs the unit tests.
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage) 
_does not apply, new package, no dependencies yet_
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by aherrmann — APPROVED** [review]
LGTM

**Comment by hzeller** [comment]
Since it is approved: Is there anything left to do before someone with permission can merge this PR ?

**Comment by ErinvanderVeen** [comment]
This PR https://github.com/NixOS/nixpkgs/pull/173225 also adds Surelog and seems to do a little more, perhaps worth to update that one instead, or close it in favour of this one.

**Review by fricklerhandwerk — APPROVED** [review]

**Review by ErinvanderVeen — APPROVED** [review]

**Comment by hzeller** [comment]
Ah, I didn't see the other PR @ErinvanderVeen, thanks. I am happy to abandon this PR, if we can get #173225 move forward instead.

**Comment by hzeller** [comment]
Confirmed it `nix-build`-ing and working on Raspberry Pi4 (aarch64-linux)

**Comment by hzeller** [comment]
Disabled Darwin for now, as running the binary seems to result in a segmentation fault on the built server, but I don't have any way to test that on my own as I don't have a Darwin machine. So changed

```patch
-    platforms = platforms.unix;
+    platforms = platforms.linux;
```

**Comment by ErinvanderVeen** [comment]
I you can get that other PR going again, I see no reason not to merge that one. Otherwise, feel free to push forward with this one instead.

**Comment by hzeller** [comment]
Hello @aherrmann @fricklerhandwerk @ErinvanderVeen , who previously approved this change.

As suggested by @ErinvanderVeen ,  I worked with the author of the more complete #173225 (which includes surelog but also the necessary other libraries) to get that change into a good shape.

I'd like you to head over there and approve that change, so that we can get that submitted instead of this change that I'll be happy to abandon then.

*TL;DR* **approve #173225 instead.**

**Comment by hzeller** [comment]
Is there anything we can do to get the other #173225 fully approved and submitted ? It feels like sometimes PRs are falling off the attention cliff...

**Comment by fricklerhandwerk** [comment]
@aherrmann can you take this?
